vkcs/compute/resource_instance_test.go

```diff
@@ -815,40 +860,89 @@ func testAccCheckComputeInstanceExists(n string, instance *servers.Server) resou
 	}
 }
 
-func testAccCheckComputeInstanceMetadata(
-	instance *servers.Server, k string, v string) resource.TestCheckFunc {
-	return func(s *terraform.State) error {
-		if instance.Metadata == nil {
-			return fmt.Errorf("No metadata")
+func testAccCheckComputeInstanceDestroyV6(s *terraform.State) error {
+	config, err := clients.ConfigureFromEnv(context.Background())
+	if err != nil {
+		return fmt.Errorf("Error authenticating clients from environment: %s", err)
+	}
+
+	computeClient, err := config.ComputeV2Client(acctest.OsRegionName)
+	if err != nil {
+		return fmt.Errorf("Error creating VKCS compute client: %s", err)
+	}
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "vkcs_compute_instance" {
+			continue
 		}
 
-		for key, value := range instance.Metadata {
-			if k != key {
-				continue
+		server, err := iservers.Get(computeClient, rs.Primary.ID).Extract()
+		if err == nil {
+			if server.Status != "SOFT_DELETED" && server.Status != "DELETED" {
+				return fmt.Errorf("Instance still exists")
 			}
+		}
+	}
 
-			if v == value {
-				return nil
-			}
+	return nil
+}
+
+func testAccCheckComputeInstanceExistsV6(resourceName string, instance *servers.Server) resource.TestCheckFunc {
```

Comment:
What's the point? Doesn't it work with SDK?

Reply:
Yes, I got an error: vkcs_cloud_monitoring resource doesn't exists. After that I change to AccTestProtoV6ProviderFactories and vkcs_cloud_monitoring start working, but test fail with panic on `acctest.AccTestProvider.Meta().(clients.Config)`. with message: interface{} is not clients.Config

Reply:
Yeah, I got it but do we really need to duplicate functions or this one where the client is configured via environment variables is enough for both providers factories?

Reply:
done